torchao/quantization/quant_api.py

```diff
@@ -630,7 +630,8 @@ def int8_dynamic_activation_int4_weight(
 
 
 def int4_weight_only(
-    group_size=128, layout=TensorCoreTiledLayout(inner_k_tiles=8), use_hqq=False
+    group_size=128, layout=TensorCoreTiledLayout(inner_k_tiles=8), use_hqq=False,
+    zero_point_domain=ZeroPointDomain.FLOAT
```

Comment:
do we need to expose this? if each layout is tied to certain settings, maybe we can do this for now: https://github.com/pytorch/ao/blob/039cef4ad546716aa04cd54c461feb173f7fe403/torchao/quantization/quant_api.py#L675

Reply:
we are also brainstorming to how to best structure these settings as well, we might be reducing some of the options here in the future

Reply:
I think layout is for weight, while zero_point_domain is for zero points. So it would be better to decouple it.
But I agree we can structure these things, better to align with the most common recipes while maintaining the capabilities to let users extend themselves.

Reply:
yeah I understand that they are separate. it's more of we are using layout as a tool to select kernels right now I think, e.g. the sparse layout example that I linked, user don't need to worry about making all the settings correct, just need to specify the sparse layout and the rest is set automatically. It is a bit confusing, but we can think about how to make it clearer in the future, one alternative I can think of is just expose a top level api for each kernel.